src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -189,8 +189,13 @@ sharded_store::project_ids(subject_schema schema) {
           return s.project_version(sub, id);
       });
 
+    const bool is_new = v_id.has_value();
+    if (is_new && schema.version != invalid_schema_version) {
+        v_id = schema.version;
+    }
```

Comment:
for the case that it isn't new, where do we check that the id is the same as the existing entry?

Reply:
I think that is in the previous commit.
Based on #9691 it looks like the expectation is to check for pre-existing schema IDs and **not** pre-existing schema versions.

Reply:
`v_id` is initialized by calling `store::project_version`, which returns `std::nullopt` if there's a version that has the same `schema_id`.